### ISSUE TITLE
Examples missing / Documentation inconsistent

### ISSUE THREAD

**Issue by YsoSirius** [mortonanalytics/myIO#12]
It would be awesome if you could include some examples in the documentation or on Github. It took me quite a while to figure out how to get a plot, how to include several layer, etc..

I didn't understand the difference between `addIoLayer` and `addIoStatLayer`, for me they do the same thing.  And the documentation of the arguments of `addIoLayer ` seem to be wrong. (**layerType, layerColor, etc..**). I am getting an error, that the argument is unused.

I wrote some small ShinyApps, which you could use as examples if you like.

```
library(myIO)
library(shiny)

data = data.frame(
  x = runif(100, 4, 100),
  y = runif(100, 60, 130),
  group = c(rep(1,50),rep(2,50))
)

ui <- fluidPage(
  myIO::myIOOutput("myoi")
)

server <- function(input, output){
  output$myoi <- myIO::renderMyIO({
    myIObj = myIO(width = "100%", height = "100%")
    
    myIO::addIoLayer(myIObj, type = "line", color = "red", data = data, label="Point-Label",
                     mapping = list(x_var = "x", y_var = "y"))
  })
}

shinyApp(ui, server)
```

Or with several Layers:
```
library(myIO)
library(shiny)

data = data.frame(
  x=1:100,
  y=1:100,
  group=c(rep(1,50),rep(2,50))
)


ui <- fluidPage(
  myIO::myIOOutput("myoi")
)

server <- function(input, output){
  output$myoi <- myIO::renderMyIO({
    myIObj = myIO(data = data, width = "100%", height = "100%")
    
    myobj1 <- myIO::addIoLayer(myIObj, type = "line", color = "red", data=data, label="ads",
                     mapping = list(x_var = "x", y_var = "y", z_var = "group"))
    
    myIO::addIoStatLayer(myobj1, type = "line", color = "blue", data = data[50:80,], label="line2",
                         mapping = list(x_var = "x", y_var = "y", z_var = "group"))
  })
}

shinyApp(ui, server)
```

Some problems I still have:

- How is the `z_var` in the mapping list used?
- Why is the vertical line in the second example only hovering where the blue points are?
- How can I define the breaks for the vertical gray line or can it be continuous too?
- How can I define individual colors, or are all points of a Layer constrained to one color only?

Some suggestions / ideas:

- It would be nice, if the color for every point could be defined, either by a color column in the dataframe or using `factor(data$color)`.
- Piping Layers would be awesome
- Being able to Enable / Disabling a layer by clicking on its label

**Comment by mortonanalytics** [comment]
Yes, I need to write a vignette (and testing scripts, and several other items to make this a complete library). Until then, you can see the demo app code here: https://github.com/mortonanalytics/shiny-server/blob/master/myio_demo/app.R.  The functions can use piping if you've loaded the magittr or dplyr packages (as you'll see in the demo).
